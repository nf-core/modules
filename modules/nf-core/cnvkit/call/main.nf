process CNVKIT_CALL {
    tag "$meta.id"
    label 'process_single'

    conda "bioconda::cnvkit=0.9.9 bioconda::samtools=1.16.1"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/cnvkit:0.9.9--pyhdfd78af_0':
        'quay.io/biocontainers/cnvkit:0.9.9--pyhdfd78af_0' }"

    input:
    tuple val(meta) , path(cns), path(vcf)

    output:
    tuple val(meta), path("*.cns"), emit: cns
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def vcf_cmd = vcf ? "-v $vcf" : ""
    """
    cnvkit.py call \\
        $cns \\
        $vcf_cmd \\
        $args \\
        -o ${prefix}.cns

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        cnvkit: \$(cnvkit.py version | sed -e 's/cnvkit v//g')
    END_VERSIONS
    """
}
