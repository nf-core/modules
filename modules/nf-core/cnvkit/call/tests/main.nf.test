
nextflow_process {

    name "Test Process CNVKIT_CALL"
    script "../main.nf"
    process "CNVKIT_CALL"

    tag "modules"
    tag "modules_nfcore"
    tag "cnvkit"
    tag "cnvkit/call"

    test("test-cnvkit-call") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cnr/test2.paired_end.sorted.cns', checkIfExists: true),
				    []
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test-cnvkit-call-with-vcf") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cnr/test2.paired_end.sorted.cns', checkIfExists: true),
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gvcf/test.genome.vcf.gz', checkIfExists: true)
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test-cnvkit-call-with-vcf-stub") {
        options '-stub'

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cnr/test2.paired_end.sorted.cns', checkIfExists: true),
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gvcf/test.genome.vcf.gz', checkIfExists: true)
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
