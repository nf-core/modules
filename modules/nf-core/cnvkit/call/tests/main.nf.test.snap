{
    "test-cnvkit-call": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cns:md5,6cff57a91d0376d5d3df6cf669935a82"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f47253e21b991f72a741d6b5c9a351a5"
                ],
                "cns": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cns:md5,6cff57a91d0376d5d3df6cf669935a82"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f47253e21b991f72a741d6b5c9a351a5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-18T17:17:47.974482513"
    },
    "test-cnvkit-call-with-vcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cns:md5,c9a2bac6fd2980071a499c0ede0d6274"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f47253e21b991f72a741d6b5c9a351a5"
                ],
                "cns": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cns:md5,c9a2bac6fd2980071a499c0ede0d6274"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f47253e21b991f72a741d6b5c9a351a5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-18T17:18:06.750975881"
    },
    "test-cnvkit-call-with-vcf-stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cns:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f47253e21b991f72a741d6b5c9a351a5"
                ],
                "cns": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cns:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f47253e21b991f72a741d6b5c9a351a5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T22:24:56.743954"
    }
}